src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -57,15 +58,32 @@ public void storeMessage(Message message) {
     List<Message> messages = new ArrayList<>();
 
     Query query =
-        new Query("Message")
-            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+            new Query("Message")
+                    .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+                    .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    return retrievedMessages(results, user, Boolean.TRUE);
+  }
+
+  public List<Message> getAllMessages(){
+    List<Message> messages = new ArrayList<>();
+
+    Query query = new Query("Message")
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    return retrievedMessages(results, "", Boolean.FALSE);
+  }
+
+  public List<Message> retrievedMessages(PreparedQuery results, String user, Boolean userGiven) {
```

Comment:
A couple things on `user` and `userGiven`:
1. It seems redundant to have a boolean parameter for `userGiven`.  The code could be simplified to just pass in the `user`, and below it could check whether `user` is an empty string rather than checking `userGiven`.
2. But actually, I think that in the case where you are passing a `user`, it will always be the case that it will be equal to the user stored in the `entity`, so maybe you don't need to pass the user at all: you can just always get `entity.getProperty("user")`.